packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/set/SetTests.kt

```diff
@@ -33,21 +34,24 @@ class SetTests {
 
     private val supportedPrimitiveTypes = TypeDescriptor.elementTypesForSet
         .filter { it.classifier != RealmObject::class } // Cannot have "pure" RealmSet<RealmObject>
-        .map {
-            (it.classifier as KClass<*>).simpleName!!
-        }
+        .map { (it.classifier as KClass<*>).simpleName!! }
+        .toSet() // Remove duplicates from nullable types
 
     private val allSupportedTypes = supportedPrimitiveTypes.plus("NonNullableSet")
 
     // ------------------------------------------------
     // RealmSet<E>
     // - supported types
-    // - unsupported type fails
+    // - RealmAny fails - mixed is always non-null
+    // - other unsupported types fails
     // ------------------------------------------------
 
     @Test
     fun `non-nullable set`() {
-        allSupportedTypes.forEach { primitiveType ->
+        // Filter out RealmAny since we cannot have RealmSet<RealmAny> - observe non-nullable type
```

Comment:
I get that, but there is no way to reason about whether this test is right or wrong since `allSupportedTypes` is already a weird set of strings and now you arbitrarily remove "RealmAny" from that 🤪 🤷 

Could we at least start defining all these subsets in the top of the file so we are grouping the definitions together in a way where it is easier to pick up the overall distinctions, reuse them and have a centralized place to look for it when we want to refactor it.